app/models/event_stream.rb

```diff
@@ -82,19 +89,30 @@ def self.group_name(group)
     group.nil? ? 'Other' : group[:name]
   end
 
+  # TODO: Consider moving since this is EmsEvent specific. group, group_level and group_name exposed as a virtual columns for reports/api.
   def group
     return @group unless @group.nil?
     @group, @group_level = self.class.group_and_level(event_type)
     @group
   end
 
+  # TODO: Consider moving since this is EmsEvent specific. group, group_level and group_name exposed as a virtual columns for reports/api.
   def group_level
     return @group_level unless @group_level.nil?
     @group, @group_level = self.class.group_and_level(event_type)
     @group_level
   end
 
+  # TODO: Consider moving since this is EmsEvent specific. group, group_level and group_name exposed as a virtual columns for reports/api.
   def group_name
     @group_name ||= self.class.group_name(group)
   end
+
+  def self.timeline_classes
+    @timeline_classes ||= EventStream.subclasses.select { |e| e.respond_to?(:group_names_and_levels) }
+  end
+
+  def self.timeline_options
+    timeline_classes.map { |c| [c.name.to_sym, c.group_names_and_levels] }.to_h
```

Comment:
Sure, I can make it private.  

I made it a method because `timeline_classes` describes what a class needs to do to implement timelines, namely, define `group_names_and_levels`.  In the example `b)`, it feels like we're mixing the concerns of "which classes support timelines" and "how to get the options for it".

Reply:
Actually, no, I think it makes sense to be able to expose what classes support timelines.